lib/ContractInteract/BrandedToken.js

```diff
@@ -411,6 +415,61 @@ class BrandedToken {
       this.contract.methods.rejectStakeRequest(stakeRequestHash),
     );
   }
+
+  /**
+   * This reedems brandedToken, must be called by beneficiary.
+   *
+   * @param {string} btToRedeem Amount of brandedTokens to redeem.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async redeem(btToRedeem, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(
+        `Invalid from address ${txOptions.from} in transaction options.`,
+      );
+      return Promise.reject(err);
+    }
+
+    const txObject = await this.redeemRawTx(btToRedeem);
+    return Utils.sendTransaction(txObject, txOptions);
+  }
+
+  /**
+   * This returns raw tx for amount to redeem.
+   *
+   * @param {string} btToRedeem Amount of brandedTokens to redeem.
+   *
+   * @return Promise<Object> Raw transaction object.
+   */
+  redeemRawTx(btToRedeem) {
+    if (!btToRedeem) {
+      const err = new TypeError(`Invalid redeemAmount: ${btToRedeem}.`);
+      return Promise.reject(err);
+    }
+
+    return Promise.resolve(
+      this.contract.methods.redeem(btToRedeem),
+    );
+  }
+
+  /**
+   * This calculates values tokens equivalent to given branded tokens.
+   *
+   * @param {string} brandedTokens Amount of branded tokens.
+   *
+   * @return {Promise<string>} Promise that resolves to amount of value tokens.
+   */
+  convertToValueTokens(brandedTokens) {
```

Comment:
this is less optimal for recurrent calls to the same BT contract; ie. once we query the conversion rate of the deployed contract, we can calculate it locally.  For now this is pre-mature optimization though; but worth marking.

Reply:
ok